python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -34,7 +35,19 @@ def __init__(
 
         self.task_definition = task_definition
         self.container_name = container_name
+
         self.secrets = secrets or []
+        if all(isinstance(secret, str) for secret in self.secrets):
+            warnings.warn(
+                "Setting secrets as a list of ARNs is deprecated. "
+                "Secrets should instead follow the same structure as the ECS API: "
+                "https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_Secret.html",
+                DeprecationWarning,
+            )
+            self.secrets = get_secrets_from_arns(self.secrets_manager, self.secrets)
+        else:
+            self.secrets = {secret["name"]: secret["valueFrom"] for secret in self.secrets}
```

Comment:
We actually transform the ECS structure right back into a dict only to later transform it back into the ECS structure. But once we drop backward compatibility, this should be easy to simplify/refactor.